api/resources_portal/views/user.py

```diff
@@ -20,14 +19,29 @@ class Meta:
         read_only_fields = ("username", "created_at", "updated_at")
 
 
+class IsUserOrAdmin(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return request.user == obj or request.user.is_superuser
+
+
 class UserViewSet(mixins.RetrieveModelMixin, mixins.UpdateModelMixin, viewsets.GenericViewSet):
     """
     Updates and retrieves user accounts
     """
 
     queryset = User.objects.all()
     serializer_class = UserSerializer
-    permission_classes = (IsUserOrReadOnly,)
+
+    # The user endpoint does not support delete or list, so we don't have to worry about permissions for these
+    def get_permissions(self):
+        if self.action == "retrieve" or self.action == "update" or self.action == "partial_update":
+            permission_classes = [IsAuthenticated, IsUserOrAdmin]
+        elif self.action == "create":
+            permission_classes = []
+        else:
+            permission_classes = [IsAuthenticated]
+
+        return [permission() for permission in permission_classes]
```

Comment:
I think that only admin users should be able to list all users. Can you add that and test it?

Reply:
yup! added and tested.